src/language/ast.ts

```diff
@@ -63,15 +82,97 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  SecurityDefinition: SecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
+}
+
+export type TypeDefinitionNode =
+  | ModelTypeDefinitionNode
+  | EnumTypeDefinitionNode
+
+export interface EndpointNamespaceTypeDefinitionNode {
+  readonly kind: 'EndpointNamespaceTypeDefinition';
+  readonly tag?: NameNode;
+  readonly description?: ModelDescriptionNode;
+  readonly endpoints?: ReadonlyArray<EndpointTypeDefinitionNode>;
+  readonly loc?: Location;
+}
+
+export interface EndpointVerbNode {
+  readonly kind: 'EndpointVerb';
+  readonly name?: NameNode;
+  readonly loc?: Location;
+}
+
+
+export interface EndpointsParameterQueryNode {
+  readonly kind: 'EndpointParameterQuery';
+  readonly name?: NameNode;
```

Comment:
in which case you will have query but don't have a name for it?

Reply:
Or

Reply:
No such case, I'll fix that